docs/guide/arrays.md

```diff
@@ -0,0 +1,79 @@
+
+| Syntax | Example |
+| :--- | :--- |
+| `ARRAYFORMULA(your_formula)` | `=ARRAYFORMULA(A2:A5*B2:B5)` |
+
+If you don't use the `ARRAYFORMULA` function, HyperFormula treats [ad-hoc arrays](#about-arrays-in-hyperformula) as [scalars](#about-arrays-in-hyperformula), taking only your ad-hoc array's top-left value.
+
+To enable array arithmetic globally, set the `useArrayArithmetic` option to `true` in HyperFormula [configuration](../api/interfaces/configparams.html#usearrayarithmetic).
+
+## Passing arrays to functions
+
+When you pass an array as a function's argument, the function returns an array as well.
+
+To pass an array as an argument, enable the array arithmetic with the [`ARRAYFORMULA`](built-in-functions.md#array-manipulation) function:
+
+| Syntax | Example |
+| :--- | :--- |
+| `YOUR_FUNCTION(ARRAYFORMULA(your_formula))` | `=ISEVEN(ARRAYFORMULA(A2:A5*10))` |
+
+If you don't use the `ARRAYFORMULA` function, HyperFormula treats [ad-hoc arrays](#about-arrays-in-hyperformula) as [scalars](#about-arrays-in-hyperformula), taking only your ad-hoc array's top-left value.
+
+To enable array arithmetic globally, set the `useArrayArithmetic` option to `true` in HyperFormula [configuration](../api/interfaces/configparams.html#usearrayarithmetic).
+
+## Constraining the output array's size
+
+To constrain the size of the output array, use the `ARRAY_CONSTRAIN` function:
+
+| Syntax | Example |
+| :--- | :--- |
+| `ARRAY_CONSTRAIN(your_array,your_height,your_width)` | `=ARRAY_CONSTRAIN(ARRAYFORMULA(A2:A5*10),2,1)` |
+
+If your specified output array size is smaller than the input array size, only the corresponding top-left cells of the input array are taken into account.
+
+If your specified output array size is larger or equal to the input array size, no change is made.
+
+## Filtering an array
+
+To filter an array based on boolean arrays, use the `FILTER` function:
+
+| Syntax | Example |
+| :--- | :--- |
+| `FILTER(your_array,BoolArray1[; BoolArray2[; ...]]` | `=FILTER(A2:A5*10),BoolArray1)` |
+
+## Array rules
+
+When operating on arrays, or passing arrays as function arguments, the following rules apply:
+
+* Array dimensions need to be consistent (e.g. every row needs to be of the same length).
+* Input array dimensions can't be larger than output array dimensions (unless you're using the [`ARRAY_CONSTRAIN`](#constraining-the-output-array-s-size) function).
+* If an input array value is missing (due to a difference in dimensions), the corresponding output array value is `#N/A`.
+* If a cell evaluates to an array, the array values are spilled into neighboring cells.<br>This behavior doesn't apply to ranges, which return a `#VALUE!` error in this case.
+* If one or both of input array dimensions is `1` (`1`x`1` or `1`x`n` or `n`x`1`), the array is repeated, to match the output array dimensions (in Google Sheets and Microsoft Excel, this behavior is different: `1`x`1` arrays are treated as [scalars](#about-arrays-in-hyperformula)).
```

Comment:
There is no inconsistence between ours, GS and XL approaches.
`1x1` is a scalar. For vectorization, dimension `1` is broadcasted to match other arguments dimension.

Reply:
@izulin - [done](https://github.com/handsontable/hyperformula/pull/741/commits/d161d5262b27709a53d9c78b9f2a529928757f76#diff-8124285aa27825ee9cad11034f206a42efb0923699c2872d25764e120df72530R78)